spec/models/vm_or_template_spec.rb

```diff
@@ -866,4 +866,155 @@
       expect(VmOrTemplate.not_archived_nor_orphaned).to eq([vm])
     end
   end
+
+  describe ".post_refresh_ems" do
+    let(:folder_blue1)   { EmsFolder.find_by(:name => "blue1") }
+    let(:folder_blue2)   { EmsFolder.find_by(:name => "blue2") }
+    let(:folder_vm_root) { EmsFolder.find_by(:name => "vm") }
+    let(:vm_blue1)       { VmOrTemplate.find_by(:name => "vm_blue1") }
+    let(:vm_blue2)       { VmOrTemplate.find_by(:name => "vm_blue2") }
+
+    let!(:ems) do
+      _, _, zone = EvmSpecHelper.local_guid_miq_server_zone
+      FactoryGirl.create(:ems_vmware, :zone => zone).tap do |ems|
+        build_vmware_folder_structure!(ems)
+        folder_blue1.add_child(FactoryGirl.create(:vm_vmware, :name => "vm_blue1", :ems_id => ems.id))
+        folder_blue2.add_child(FactoryGirl.create(:vm_vmware, :name => "vm_blue2", :ems_id => ems.id))
+      end
+    end
+
+    let!(:start_time) { Time.now.utc }
```

Comment:
The start time has to kick before each of the methods begin, otherwise it won't be accurate